konan_sdk/sdk.py

```diff
@@ -19,88 +18,30 @@ def __init__(self, auth_url="https://auth.konan.ai", api_url="https://api.konan.
         self.password = ''
 
         self.user: Optional[KonanUser] = None
+        self.auth: Optional[KonanAuth] = None
 
         if not verbose:
             logger.remove()
             logger.add(sys.stderr, level="INFO")
 
     def login(self, email: str, password: str) -> KonanUser:
-        # Save user credentials in case we need to login again
-        self.email = email
-        self.password = password
-
-        # Login using provided credentials
-        response = requests.post(
-            self.auth_url + LOGIN_ENDPOINT,
-            data={
-                'email': email,
-                'password': password
-            }
-        )
-        # Raise any HTTP errors > 400
-        # TODO: Check for 401, would mean invalid credentials were supplied
-        response.raise_for_status()
-
-        # Extract authentication tokens
-        tokens = response.json()
-
-        # Create a new user using these token
-        self.user = KonanUser(tokens['access'], tokens['refresh'])
-
-        logger.info(f"Successfully logged in using {self.email}")
-        return self.user
 
-    def _post_login_checks(self) -> None:
-        assert self.user is not None, "User credentials were not provided. Please use the .login() function first."
+        self.auth = KonanAuth(email=email, password=password, auth_url=self.auth_url)
+        self.user = self.auth.login()
+        return self.user
 
-    def refresh_token(self) -> None:
-        self._post_login_checks()
+    def predict(self, deployment_uuid: str, input_data: Union[Dict, str]) -> Tuple[str, Dict]:
 
-        logger.debug("Sending token refresh request.")
-        response = requests.post(
-            self.auth_url + TOKEN_REFRESH_ENDPOINT,
-            data={
-                "refresh": self.user.refresh_token
-            },
+        # check user performed login
+        self.auth._post_login_checks()
 
-        )
-        logger.debug("Received token refresh response. Parsing output.")
+        # Check if access token is valid and retrieve a new one if needed
+        self.auth.auto_refresh_token()
 
-        response.raise_for_status()
+        prediction_uuid, output = PredictionEndpoint(api_url=self.api_url, user=self.user, deployment_uuid=deployment_uuid).post(payload=input_data)
```

Comment:
Use the `KonanUser` object inside the `KonanAuth` object instead.
```suggestion
        prediction_uuid, output = PredictionEndpoint(api_url=self.api_url, user=self.auth.user, deployment_uuid=deployment_uuid).post(payload=input_data)
```